field/TinyLang/Field/Generator.hs

```diff
@@ -384,10 +421,34 @@ defaultUniConst =
     where
         uni = knownUni @f @a
 
--- We do not provide an implementation for 'arbitrary' (because we don't need it and it'd be
--- annoying to write it), but we still want to make provide an 'Arbitrary' instance, so that
--- 'shrink' can be used in the 'Arbitrary' instance of 'Expr' (a separately provided
--- 'shrinkStatement' wouldn't work, because we want to shrink a pair of values, see the instance).
+instance (Field f, Arbitrary f) => Arbitrary (Program f) where
+    arbitrary = mkProgram <$> arbitrary
+    shrink    = fmap mkProgram . shrink . unProgram
+
+instance (Field f, Arbitrary f) => Arbitrary (Statements f) where
+    arbitrary = runSGen vars stmtsGen where
+        vars = defaultVars
+        stmtsGen = sized $ \size -> do
+            adjustUniquesForVars vars
+            boundedArbitraryStmts size
+
+    shrink stmts = mkStatements <$> preservingShrink stmts' ++ nonPreservingShrink stmts' where
+        stmts' :: [Statement f]
+        stmts' = unStatements stmts
```

Comment:
Again would be nice to use simple pattern matching instead of introducing another binding.

Reply:
Hmm, actually
`shrink stmts | stmts' <- unStatements stmts = .... where`
does not bind `stmts'` in the `where` clause, which I find surprising.

Wonder if that is on purpose...